### DISCUSSION TITLE
Simplification of YOLOv5 architecture

### DISCUSSION THREAD

**Discussion by lessleeb** [ultralytics/yolov5#6691] [Q&A]
Hello, I have a specific question regarding YOLOv5 architecture.
Unfortunately, I am not very familiar with the design of a neural network and the function of every layer. 
I would like to detect simple geometries such as small circular or squared objects in an image, I think it is not necessary to use the whole architecture for this task.
So my question is how could I modify the architecture to simplify the size of the model and which layers are important for this simple detection and which layers could be eliminated?
I wonder if I simplify the model, will my inference time get reduced as well?
Sorry for the specific question but I hope I can get some guidance from people who know about this. Thank you!

**Comment by glenn-jocher** [comment]
@lessleeb YOLOv5 comes in a variety of different sizes just for this purpose. The sizes are created simply using different compound scaling constants. If you're task is easier just use a smaller model.

<img width="856" alt="Screenshot 2022-02-18 at 13 29 52" src="https://user-images.githubusercontent.com/26833433/154683163-08806035-fc48-4b42-a11e-6399d20b9366.png">
